build failure because of missing PYTHON_LIBRARIES

This is failing Grok pipelines.

```
Cloning into 'nta'...
Python 2.7.5 :: Anaconda 1.8.0 (64-bit)
CMake Error at /usr/share/cmake28/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find PythonLibs (missing: PYTHON_LIBRARIES) (found suitable
  version "2.7.5", minimum required is "2.7")
Call Stack (most recent call first):
  /usr/share/cmake28/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake28/Modules/FindPythonLibs.cmake:186 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:378 (FIND_PACKAGE)


make: *** No targets specified and no makefile found.  Stop.

STDOUT redirected to: /opt/jenkinshome/jobs/nupic-pipeline/workspace/2014-03-08_00-31-47RILAOA/nupic/build_system/stdout.txt
Build failed!!!
```

Happened after merge of https://github.com/numenta/nupic/pull/717.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -340,17 +340,7 @@ message(STATUS "  Python embedding link flags: ${NTA_PLATFORM_LINKFLAGS_PYEMBED}
 # Instead we just use a shell command to execute a simple python command, if exit code is 0 ('success') then we have python installed.
 #
 message(STATUS "Checking software tools...")
-
-#
-# Check for user-defined PYTHON env variable
-#
-if(NOT "$ENV{PYTHON}" STREQUAL "")
-  set(PYTHON "$ENV{PYTHON}")
-else()
-  set(PYTHON "python")
-endif()
-
-execute_process(COMMAND ${PYTHON} --version
+execute_process(COMMAND python --version
                 RESULT_VARIABLE EXIT_CODE)
 if(NOT EXIT_CODE EQUAL 0)
   message(FATAL_ERROR "System Python not found. You do not have a system version of Python or it is not set on environment path.")
@@ -360,7 +350,7 @@ endif()
 # Gives the version of Python necessary to get installation directories
 # for use with ${PYTHON_VERSION}, etc.
 #
-execute_process(COMMAND ${PYTHON} -c "import sys;sys.stdout.write(str(sys.version_info[0]) + '.' + str(sys.version_info[1]))"
+execute_process(COMMAND python -c "import sys;sys.stdout.write(str(sys.version_info[0]) + '.' + str(sys.version_info[1]))"
                 OUTPUT_VARIABLE PYTHON_VERSION)
 if(NOT(${PYTHON_VERSION} EQUAL "2.6" OR ${PYTHON_VERSION} EQUAL "2.7"))
   message(FATAL_ERROR "Only these versions of Python are accepted: 2.6, 2.7")
@@ -369,21 +359,20 @@ endif()
 #
 # Find out where system installation of python is.
 #
-execute_process(COMMAND ${PYTHON} -c "import sys;sys.stdout.write(sys.prefix)"
+execute_process(COMMAND python -c "import sys;sys.stdout.write(sys.prefix)"
                 OUTPUT_VARIABLE PYTHON_PREFIX)
 string(REGEX REPLACE "\\\\" "/" PYTHON_PREFIX ${PYTHON_PREFIX})
 set(PYTHON_INCLUDE_DIR "${PYTHON_PREFIX}/include/python${PYTHON_VERSION}")
 set(PYTHON_LIBRARY_DIR "${PYTHON_PREFIX}/lib/python${PYTHON_VERSION}")
 set(PYTHON_SITE_PACKAGES_DIR "${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION}/site-packages")
-FIND_PACKAGE(PythonLibs ${PYTHON_VERSION} REQUIRED)
 
 #
 # Finds out version of Numpy and headers's path.
 #
-execute_process(COMMAND ${PYTHON} -c "import sys;import numpy;sys.stdout.write(numpy.get_include())"
+execute_process(COMMAND python -c "import sys;import numpy;sys.stdout.write(numpy.get_include())"
                 OUTPUT_VARIABLE NUMPY_INCLUDE_DIR)
 string(REGEX REPLACE "\\\\" "/" NUMPY_INCLUDE_DIR ${NUMPY_INCLUDE_DIR})
-execute_process(COMMAND ${PYTHON} -c "import sys;import numpy;sys.stdout.write(numpy.__version__)"
+execute_process(COMMAND python -c "import sys;import numpy;sys.stdout.write(numpy.__version__)"
                 OUTPUT_VARIABLE NUMPY_VERSION)
 
 #
@@ -405,8 +394,6 @@ message(STATUS "  Version: ${PYTHON_VERSION}")
 message(STATUS "  Include dir: ${PYTHON_INCLUDE_DIR}")
 message(STATUS "  Library dir: ${PYTHON_LIBRARY_DIR}")
 message(STATUS "  Site packages dir: ${PYTHON_SITE_PACKAGES_DIR}")
-message(STATUS "  Python lib: ${PYTHON_LIBRARY}")
-message(STATUS "  Python command: ${PYTHON}")
 message(STATUS "Numpy details:")
 message(STATUS "  Version: ${NUMPY_VERSION}")
 message(STATUS "  Include dir: ${NUMPY_INCLUDE_DIR}")
@@ -995,7 +982,6 @@ set(LIBRARIES
   ${LIB_STATIC_UTILS}
   ${LIB_STATIC_TYPES}
   ${NTA_STANDARD_LIBS}
-  ${PYTHON_LIBRARY}
 )
 generate_dynamic_library(${LIB_DYNAMIC_CPPREGION} "${NTA_CXXFLAGS}" "${SOURCE}" "${NTA_LINKFLAGS} ${NTA_LINKFLAGS_PYEMBED}" "${LIBRARIES}")
 

